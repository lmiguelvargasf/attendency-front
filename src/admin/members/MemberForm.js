import React from 'react'
import { Form, Input, Button } from 'antd'

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 8 }
}
const tailLayout = {
  wrapperCol: { offset: 8, span: 8 }
}

const MemberForm = ({ member, onFinish, onFinishFailed }) => {
  const inputs = [
    { label: 'First Name', name: 'firstName', rules: [{ required: true, message: "Please input member's first name!" }] },
    { label: 'Middle Name', name: 'middleName' },
    { label: 'Last Name', name: 'lastName', rules: [{ required: true, message: "Please input member's last name!" }] },
    { label: 'Preferred Name', name: 'preferredName' },
    { label: 'Email', name: 'email', rules: [{ type: 'email', message: 'Invalid email!' }, { required: true, message: "Please input member's email!" }] }
  ]

  return (
    <Form
      data-testid='create-member-form'
      {...layout}
      name='member'
      initialValues={member}
      onFinish={onFinish}
      onFinishFailed={onFinishFailed}
    >
      <>
        {inputs.map((input, index) => (
          <Form.Item key={index} {...input}>
            <Input />
          </Form.Item>
        ))}
      </>
      <Form.Item {...tailLayout}>
        <Button type='primary' htmlType='submit' data-testid='create-member-button'>
          {member ? 'Save' : 'Create'}
        </Button>
      </Form.Item>
    </Form>
  )
}

export default MemberForm
