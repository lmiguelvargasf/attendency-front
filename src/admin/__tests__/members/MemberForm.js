import React from 'react'
import { render, cleanup } from '@testing-library/react'
import MemberForm from '../../members/MemberForm'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

describe('MemberForm component', () => {
  afterEach(cleanup)

  describe('member is passed', () => {
    let component
    beforeEach(() => {
      const member = {
        firstName: 'John',
        middleName: 'Patrick',
        lastName: 'Smith',
        preferredName: 'JP',
        email: 'john.smith@example.com'
      }
      component = (
        <MemberForm member={member} onFinish={jest.fn()} onFinishFailed={jest.fn()} />
      )
    })
    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays button saying Save', () => {
      const { getByTestId } = render(component)
      expect(getByTestId('create-save-button')).toHaveTextContent(/^Save$/)
    })
  })

  describe('member is not passed', () => {
    let component

    beforeEach(() => {
      component = (
        <MemberForm onFinish={jest.fn()} onFinishFailed={jest.fn()} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays button saying Create', () => {
      const { getByTestId } = render(component)
      expect(getByTestId('create-save-button')).toHaveTextContent(/^Create$/)
    })
  })
})
