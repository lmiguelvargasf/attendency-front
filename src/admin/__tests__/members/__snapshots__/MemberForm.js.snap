// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberForm component member is not passed matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    data-testid="create-member-form"
    id="member"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="member_firstName"
          title="First Name"
        >
          First Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_firstName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="member_middleName"
          title="Middle Name"
        >
          Middle Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_middleName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="member_lastName"
          title="Last Name"
        >
          Last Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="member_preferredName"
          title="Preferred Name"
        >
          Preferred Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_preferredName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="member_email"
          title="Email"
        >
          Email
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_email"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              data-testid="create-save-button"
              type="submit"
            >
              <span>
                Create
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`MemberForm component member is passed matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    data-testid="create-member-form"
    id="member"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="member_firstName"
          title="First Name"
        >
          First Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_firstName"
              type="text"
              value="John"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="member_middleName"
          title="Middle Name"
        >
          Middle Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_middleName"
              type="text"
              value="Patrick"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="member_lastName"
          title="Last Name"
        >
          Last Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_lastName"
              type="text"
              value="Smith"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="member_preferredName"
          title="Preferred Name"
        >
          Preferred Name
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_preferredName"
              type="text"
              value="JP"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="member_email"
          title="Email"
        >
          Email
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              id="member_email"
              type="text"
              value="john.smith@example.com"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              data-testid="create-save-button"
              type="submit"
            >
              <span>
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
